Project 2 - first submition

olggaf opened the pull request:
Project 2 - first submition - sending for a review

olggaf committed:
Project 2 - first submit

Project 2 - first submit

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+
+.overlay {
+  background-color: rgba(0, 0, 0, 0.6);
+  inset: 0 0;
```
---

**Отлично:**

Корректное выравнивание

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+      </nav>
+      <!-- logo -->
+      <a class="header__logo"
+        ><img class="header__logo-image" src="./images/logo-black.svg" alt=""
```
- Альтернативное описание для тега img должно быть на языке страницы(русский). Альтернативное описание картинки должно объяснять что находится на этой картинке. Оно часто используется людьми с проблемами зрения, которые используют специальные читалки, где все картинки будут   описаны через альтернативный текст. Также хорошее описание картинок помогает для индексации страницы поисковыми системами.    Исключения: логотипы английских брендов. Например у иконки Instagram допустим alt на английском.

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>Оно тебе надо — аукцион вещей, в которые никто не верил</title>
+    <link rel="stylesheet" href="./fonts/fonts.css" />
+    <link rel="stylesheet" href="./styles/global.css" />
```
---

**Отлично:**

Правильный порядок подключения стилей. Шрифты должны быть подключены первыми, потом общие стили(сброс), потом кастомные. При таком порядке подключении шрифты могут быть использованы в двух последующих файлах.  А так же сброс стилей должен быть до подключения своих стилей из-за специфики CSS, чтобы `global.css` не переопределил никаких наших стилей.

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        </ul>
+      </nav>
+      <!-- logo -->
+      <a class="header__logo"
```
---

**Отлично:**

Логотип должен быть ссылкой без атрибута href(что бы по ссылке не было перехода)  для улучшения семантики страницы.

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        ><img class="header__logo-image" src="./images/logo-black.svg" alt=""
+      /></a>
+      <!-- contacts -->
+      <address class="address header__address">
```
---

**Отлично:**

Отличная семантика

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        </h1>
+        <div class="cover__description">
+          <!-- subheading -->
+          <h2 class="cover__description-text">
```
Это не заголовок, используйте тег p

Avaakum replied:
Не исправлено

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+      </section>
+      <!-- lots -->
+      <section class="lots">
+        <h3 class="lots__heading">Лоты</h3>
```
- Уровень заголовка не соответствует структуре страницы. Это заголовок секции, по структуре будет  h2

---


- Про заголовки и подзаголовки на примере другой страницы с секциями, хэдером, и футером: у нас есть **один заголовок h1 на странице**, в каждой секции есть свой заголовок h2, если там есть другие элементы с заголовками ниже по иерархии, то они будут h3. Если например в секции был бы заголовок h2 и подзаголовок h3, то элементы с заголовками ниже по иерархии в этой же секции были бы уже h4.

Avaakum replied:
Не исправлено

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+            <a href="#0" class="card-link">
+              <article class="card">
+                <div class="overlay"></div>
+                <h4 class="card__title">
```
- Уровень заголовка не соответствует структуре страницы. Тут  и далее в секции будут h3 по структуре

Avaakum replied:
Не исправлено

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        <h3 class="lots__heading">Лоты</h3>
+        <ul class="lots__card-list">
+          <li class="lots__card-list-item card_type_film">
+            <a href="#0" class="card-link">
```
---

**Отлично:**

Отличная семантика и структура

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+          </div>
+        </div>
+        <!-- about article -->
+        <article class="about__column">
```
---

**Отлично:**

Отличная семантика

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        </div>
+        <!-- about article -->
+        <article class="about__column">
+          <h3 class="about__title">Об аукционе</h3>
```
- Уровень заголовка не соответствует структуре страницы. Это заголовок секции, по структуре будет  h2

Avaakum replied:
Не исправлено

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+        </ul>
+      </nav>
+      <!-- social media links -->
+      <ul class="footer__social-list">
```
---

**Можно лучше:**

Можно использовать для обертки списка ссылок тег `<nav>`, можно даже без класса, только для семантики.
Этим элементом принято выделять список из ссылок, отвечающий за навигацию по сайту и не только. Добавление этого элемента улучшает восприятие вашей верстки поисковыми машинами, а так же скринридерами. 
Можно использовать несколько тегов `nav` в проекте.

https://doka.guide/html/nav/#alyona-batickaya-sovetuet

Avaakum commented on index.html:
```diff
@@ -5,6 +5,207 @@
+      </nav>
+      <!-- social media links -->
+      <ul class="footer__social-list">
+        <li class="footer__social-list-item">
```
---

**Отлично:**

Отлично, использован список для повторяющихся элементов

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+/* header */
+.header {
+  display: grid;
+  grid-template-columns: 1fr 132px 1fr;
```
---

**Можно лучше:**

В header для выравнивания удобно использовать грид. Ячейки по краям будут занимать ширину по одному фрагменту, а логотип только ширину равную минимальной ширине контента (min-content) ` 
    grid-template-columns: 1fr min-content 1fr; `  .  
  
  А для выравнивания внутри грид-ячейки в таких случаях удобно использовать

- https://developer.mozilla.org/en-US/docs/Web/CSS/justify-self
- https://developer.mozilla.org/ru/docs/Web/CSS/align-self

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+  margin-left: 30px;
+  word-spacing: 0.62em;
+}
+.bet-button {
```
https://disk.yandex.ru/i/ZY31QVMOsyMthw кнопка не наследует семейство шрифтов, необходимо задать либо ` font-family:inherit; ` (что бы унаследовала), либо задать само значение ` font-family: Raleway, sans-serif; `

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+  column-gap: 25px;
+  justify-content: center;
+  color: #fff;
+  flex-wrap: wrap;
```
---

**Отлично:**

Отлично! При добавлении новых карточек они переносятся на новую строку и растягиваются на всю ширину родительского контейнера

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+  display: flex;
+  list-style: none;
+  column-gap: 25px;
+  justify-content: center;
```
https://disk.yandex.ru/i/v4ApG0VaFny47w

- используйте значение flex-start, что бы при добавлении новых карточек они начинались от левого края
- Используйте свойство gap между элементами списка для задания отступ, что бы при добавлении новых карточек такой же отступ между ними был и по горизонтали

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+  /* letter-spacing: -1px; */
+}
+
+.lots__look-more-link {
```
---

**Можно лучше:**

Нужно ограничить максимальную ширину ссылки в соответствии с контентом, значение `fit-content` или изменить выравнивание элемента, чтобы он не занимал больше пространства, чем внутренний контент. Сейчас ссылка занимает всю ширину родителя и может быть нажата в пустом месте.

https://developer.mozilla.org/ru/docs/Web/CSS/max-width

Avaakum commented on styles/style.css:
```diff
@@ -1 +1,234 @@
+  margin-bottom: 25px;
+  /* border: 1px solid darkmagenta; */
+}
+.about__text:last-of-type {
```
---

**Отлично:**

Круто, что уже знаете псевдо-классы

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Здравствуйте!  У вас получилась хорошая работа!

_Что понравилось:_

- Хорошая структура HTML
- похоже на макет
- нет скролла от 1100px и более и все элементы отображаются корректно
- Хорошее форматирование документа
- хорошие стили
- используется семантика
- Используются списки


_Что необходимо исправить:_

- **Все комментарии, кроме комментариев с пометками «Можно лучше» и «Отлично». А так же не забудьте проверить, есть ли свёрнутые комментарии, которые обычно находятся по центру страницы с ревью при прокрутке вниз, и выглядят следующим образом https://disk.yandex.ru/i/FyX-8tzeqLjimw**





_Что можно улучшить:_

- Все комментарии "Можно лучше"

- Селекторы в файле стилей должны располагаться в порядке их появления в html

- В любом месте в файлах CSS в проекте(кроме самого CSS файла подключения шрифтов, где указывается только название подключаемого шрифта), где указано свойство  `font-family`  помимо основного шрифта так же должен быть указан альтернативный шрифт и тип засечек. Это делается для того, что если по каким-то причинами основной шрифт не загрузится, то внешний вид страницы не будет изменен очень сильно.  Порядок написания шрифтов: Основной шрифт, [Веб-безопасный альтернативный шрифт(по ссылке список этих шрифтов)](https://www.cssfontstack.com/), Тип засечек. Пример ` font-family: Inter, Arial, sans-serif; ` 

- Желательно использовать все классы по БЭМ. 
  - https://ru.bem.info/methodology/quick-start/
  - Проверить ошибки БЭМ можно на  https://nglazov.github.io/bem-validator-page/
  - струтуру можно проверить на https://yoksel.github.io/html-tree/

- Можно использовать атрибуты `target="_blank"` для ссылок на сторонние ресурсы для того чтобы пользователю не приходилось заново открывать сайт проекта
- Для более точной корректировки размеров текстов и расстояний между элементами можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi) для Chrome 


**Напоминание:** После выполнения всех критических замечаний работа будет принята. Предложения по улучшению выполняются по желанию.

Желаю успеха!

olggaf committed:
Project 2 - Second submission

Project 2 - Second submission

Avaakum reviewed (changes requested):
Развернуть общий комментарий ⬇︎

Здравствуйте. К сожалению проект отклоняется от проверки. Итерация не теряется. 




- _На повторных итерациях не исправлены критические замечания._ 
    - **Пример:** 
              - Комментарии в html коде  в ревью предыдущей итерации. Если у вас есть сомнения по поводу того, что комментарий исправлен, то, пожалуйста, перечитайте комментарии внимательно, там описано, что нужно исправить в работе. Если что-то исправить не получается, то можно смело обращаться к наставнику, он сможет подсказать. Чтобы исправить ошибки валидатора(при наличии), можно в chrome перевести страницу на русский, чтобы было понятнее, что именно за ошибки он выдаёт. А для соответствия макету можно воспользоваться [расширением PixelPerfect](https://chrome.google.com/webstore/detail/perfectpixel-by-welldonec/dkaagdgjmgdmbnecmcefdhjekcoceebi  )  для Chrome.



     - Пожалуйста, будьте внимательны, проверяйте, исправляйте и присылайте работу только после исправления **всех критических замечаний**  с предыдущей итерации
    - Если у вас есть технические проблемы, то попробуйте обратиться в поддержку или спросить в чате студентов не было ли у кого похожих случаев.

olggaf committed:
Project 2 - 3rd submission

Прошу прощения, в предыдущей итерации не увидела комментарии, спрятанные под многоточием, в след. раз буду знать об этом. Все исправила.

Avaakum reviewed (approved):
Развернуть общий комментарий ⬇︎

Здравствуйте! Рад вам сообщить, что ваша работа принята. Сейчас страничка выглядит прекрасно, всё работает отлично. Отличная семантика и структура. Желаю вам удачи в следующих спринтах и проектах! 



Ссылки:

- Можно использовать неразрывные пробелы для того чтобы текст не переносился в неожиданных местах. Неразрывные пробелы заменяют обычные, если нужно чтобы какая-то фраза не переносилась, например не нужно чтобы "на собаках" раскидывалось по разным строчкам при некоторых ширинах экранов, тогда `«на&nbsp;собаках»`. Но если страница с адаптацией, то всегда стоит проверять как текст с пробелами выглядит на всех ширинах.
- Про альтернативный текст для картинок https://habr.com/ru/post/341810/
- [Ссылка про подгрузку шрифтов](https://only-to-top.ru/blog/tools/2019-03-13-nastrojte-pokaz-vsego-teksta-vo-vremya-zagruzki-veb-shriftov.html) 



Несколько  ссылок на материал по теме нативных CSS переменных от авторов практикума:

- [1 часть](https://tproger.ru/articles/custom-properties-in-css-part-1/)

- [2 часть](https://tproger.ru/articles/custom-properties-in-css-part-2/)

- [3 часть](https://tproger.ru/articles/custom-properties-in-css-part-3/)
